### PR TITLE
[TOOL-1339] Add metrics

### PR DESCRIPTION
**Pull request by Fauzyy**
Instrument each step: plan, apply, init, and run

### CODE REVIEW

File: server/events/command_metrics.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+func InstrumentStepRunner(runner StepRunner, eng *stats.Engine, step string) StepRunner {
+	return &instrumentedStepRunner{
+		stats:  eng.WithPrefix("atlantis"),
+		runner: runner,
+		step:   step,
+	}
+}
+
+// InstrumentCustomRunner wraps step runners to provide metrics for: run steps
+func InstrumentCustomRunner(runner CustomStepRunner, eng *stats.Engine) CustomStepRunner {
```
Comment (Fauzyy) [inline server/events/command_metrics.go:32]: `run` steps have their own step runner interface so we need two 😢

File: server/events/command_metrics.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		ic.stats.Incr("steps.error", tags...)
+		return out, err
+	}
+	ic.stats.ClockAt("steps.duration", start, tags...).Stop()
+	ic.stats.Incr("steps.success", tags...)
+	return out, err
+}
+
+// TODO: Make this list a configurable map
+func errorTag(output string) string {
```
Comment (Fauzyy) [inline server/events/command_metrics.go:81]: Would be nice to pass this in as a config map, but not necessary right now

File: server/events/command_metrics_test.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	ic.Run(projectContext, nil, "/path")
+
+	found := measures(t, eng)
+	if len(found) != 2 {
+		t.Fatalf("expected 2 measures got %d", len(found))
+	}
+
+	checkHistogramEqual(t, found[0], stats.Measure{
+		Name:   "test.atlantis.steps.duration",
+		Fields: []stats.Field{stats.MakeField("", 1, stats.Histogram)},
```
Comment (Fauzyy) [inline server/events/command_metrics_test.go:68]: I'm not sure why this shows up as empty 🤷 @eculver

Comment (eculver) [reply server/events/command_metrics_test.go:68]: I think it could be because the prefix hasn't been applied? Not sure though.

**Pull request merged**